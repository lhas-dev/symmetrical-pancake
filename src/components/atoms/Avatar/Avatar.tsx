import styled from "styled-components";
import { useDevice } from "../../../hooks/useDevice";

interface AvatarProps {
  src?: string;
  name?: string;
  hasNotification?: boolean;
}

const Wrapper = styled.button`
  margin: 0;
  padding: 0;
  display: flex;
  align-items: center;
  background: transparent;
  border: none;
  cursor: pointer;
  position: relative;
`;

const Image = styled.img`
  width: 28px;
  height: 28px;
  border-radius: 99px;

  @media (min-width: 768px) {
    width: 36px;
    height: 36px;
  }
`;

const Badge = styled.span`
  width: 8px;
  height: 8px;
  background: #e72176;
  border-radius: 8px;
  display: inline-block;
  position: absolute;
  right: -1px;
  top: -1px;
`;

export const Avatar = ({ src, name, hasNotification }: AvatarProps) => {
  const device = useDevice();
  return (
    <Wrapper>
      {hasNotification && device.isMobile() && <Badge />}
      <Image src={src} alt={name} />
    </Wrapper>
  );
};
