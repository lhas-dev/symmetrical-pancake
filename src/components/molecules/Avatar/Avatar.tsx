import styled from "styled-components";
import { useDevice } from "../../../hooks/useDevice";
import { Badge } from "../../atoms/Badge/Badge";

interface AvatarProps {
  src?: string;
  name?: string;
  hasNotification?: boolean;
}

const Wrapper = styled.button`
  margin: 0;
  padding: 0;
  display: flex;
  align-items: center;
  background: transparent;
  border: none;
  cursor: pointer;
  position: relative;

  ${Badge} {
    position: absolute;
    right: -1px;
    top: -1px;
  }
`;

const Image = styled.img`
  width: 28px;
  height: 28px;
  border-radius: 99px;

  @media (min-width: 768px) {
    width: 36px;
    height: 36px;
  }
`;

export const Avatar = ({ src, name, hasNotification }: AvatarProps) => {
  const device = useDevice();
  return (
    <Wrapper>
      {hasNotification && device.isMobile() && <Badge />}
      <Image src={src} alt={name} />
    </Wrapper>
  );
};
