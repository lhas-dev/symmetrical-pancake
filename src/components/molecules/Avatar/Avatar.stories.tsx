import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Avatar } from "./Avatar";

export default {
  title: "Storybook/Molecules/Avatar",
  component: Avatar,
  argTypes: {
    src: { control: "text" },
    name: { control: "text" },
    hasNotification: { control: "boolean" },
  },
} as ComponentMeta<typeof Avatar>;

const Template: ComponentStory<typeof Avatar> = (args) => <Avatar {...args} />;

export const Default = Template.bind({});
Default.args = {
  src: "https://picsum.photos/200",
  name: "John Doe",
  hasNotification: false,
};
