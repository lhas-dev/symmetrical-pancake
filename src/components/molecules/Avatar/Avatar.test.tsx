import { render, act } from "@testing-library/react";
import { Avatar } from "./Avatar";

test("renders Avatar with valid params on mobile devices", async () => {
  Object.defineProperty(window.navigator, "userAgent", {
    value: "Android",
    writable: true,
  });

  const tree = await act(async () => {
    render(
      <Avatar src="https://picsum.photos/200" name="John Doe" hasNotification />
    );
  });
  expect(tree).toMatchSnapshot();
});

test("renders Avatar with valid params on desktop devices", async () => {
  Object.defineProperty(window.navigator, "userAgent", {
    value: "Chrome",
    writable: true,
  });

  const tree = await act(async () => {
    render(
      <Avatar src="https://picsum.photos/200" name="John Doe" hasNotification />
    );
  });
  expect(tree).toMatchSnapshot();
});
