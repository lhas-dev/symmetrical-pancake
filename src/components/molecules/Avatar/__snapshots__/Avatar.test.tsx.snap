// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Avatar with valid params on desktop devices 1`] = `undefined`;

exports[`renders Avatar with valid params on mobile devices 1`] = `undefined`;
